makeflow/combined_workflow.jx

```diff
@@ -0,0 +1,250 @@
+{
+  "define": {
+    "ARCHIVE_FILE_PATH": ARCHIVE_DIR + ARCHIVE_FILE,
+    "METADATA": ARCHIVE_DIR + UUID + "_metadata.json",
+    "LEFT_BIN": ARCHIVE_DIR + UUID + "_left.bin",
+    "RIGHT_BIN": ARCHIVE_DIR + UUID + "_right.bin",
+    "METADATA_CLEANED": CLEANED_META_DIR + UUID + "_metadata_cleaned.json",
+    "LEFT_TIF": TIFS_DIR + UUID + "_left.tif",
+    "RIGHT_TIF": TIFS_DIR + UUID + "_right.tif",
+    "LEFT_SOILMASK": SOILMASK_DIR + UUID + "_left_mask.tif",
+    "RIGHT_SOILMASK": SOILMASK_DIR + UUID + "_right_mask.tif",
+    "MOSAIC_LIST_FILE": FIELDMOSAIC_DIR + "filelist.txt"
+  },
+  "rules": [
+    {
+      # Make directory to download archive file into
+      "command": "mkdir -p ${ARCHIVE_DIR}",
+      "environment": {
+        "ARCHIVE_DIR": ARCHIVE_DIR
+      },
+      "inputs": [],
+      "outputs": [
+        ARCHIVE_DIR
+      ]
+    },
+    {
+      # Download archive file
+      "command": "wget -O ${ARCHIVE_FILE_PATH} ${ARCHIVE_URL}",
+      "environment": {
+        "ARCHIVE_DIR": ARCHIVE_DIR,
+        "ARCHIVE_FILE": ARCHIVE_FILE,
+        "ARCHIVE_URL": ARCHIVE_URL,
+        "ARCHIVE_FILE_PATH": ARCHIVE_FILE_PATH
+      },
+      "inputs": [
+        ARCHIVE_DIR
+      ],
+      "outputs": [
+        ARCHIVE_FILE_PATH
+      ]
+    },
+    {
+      # Extract the archive
+      "command": "unzip -o ${ARCHIVE_FILE_PATH} -d ${ARCHIVE_DIR}",
+      "environment": {
+        "ARCHIVE_DIR": ARCHIVE_DIR,
+        "ARCHIVE_FILE_PATH": ARCHIVE_FILE_PATH
+      },
+      "inputs": [
+        ARCHIVE_DIR,
+        ARCHIVE_FILE_PATH
+      ],
+      "outputs": [
+        METADATA,
+        LEFT_BIN,
+        RIGHT_BIN
+      ]
+    },
+    {
+      # Make a cleaned copy of the metadata
+      "command": "mkdir -p ${WORKING_SPACE} && docker run --rm -e BETYDB_URL=${BETYDB_URL} -e BETYDB_KEY=${BETYDB_KEY} -v $(pwd):/mnt -w /mnt agpipeline/cleanmetadata:latest --metadata ${METADATA} --working_space ${WORKING_SPACE} ${SENSOR} ${USERID}",
+      "environment": {
+        "BETYDB_URL": BETYDB_URL,
+        "BETYDB_KEY": BETYDB_KEY,
+        "SENSOR": "stereoTop",
+        "METADATA": METADATA,
+        "WORKING_SPACE": CLEANED_META_DIR,
+        "USERID": ""
+      },
+      "inputs": [METADATA],
+      "outputs": [
+        CLEANED_META_DIR,
+        METADATA_CLEANED
+      ]
+    },
+    {
+      # Make directory to store TIFF files
+      "command": "mkdir -p ${TIFS_DIR}",
+      "environment": {
+        "TIFS_DIR": TIFS_DIR
+      },
+      "inputs": [],
+      "outputs": [
+        TIFS_DIR
+      ]
+    },
+    {
+      # Convert LEFT bin/RGB image to TIFF format
```

Comment:
Two questions: can we make this so that it doesn't need left **and** right, if that makes sense; is there a way to make the left and right processing a loop (if both are needed here)?

Reply:
Yes, there is a way. It used to be like this. As I recall I then ran into a problem with inputs & dependencies downstream, and this was the simplest way to get it working again.

Reply:
New issue: Make the 'left' and 'right' processing a loop #51